templates/_worker.js/utils/http.ts

```diff
@@ -0,0 +1,82 @@
+import type { MatchPCREResult } from './pcre';
+import { applyPCREMatches } from './pcre';
+
+/**
+ * Applies a set of headers to a response.
+ *
+ * @param source Headers to apply.
+ * @param target Headers object to apply to.
+ * @param pcreMatch PCRE match result to apply to header values.
+ */
+export function applyHeaders(
+	source: Record<string, string> | Headers,
+	target: Headers,
+	pcreMatch?: MatchPCREResult
+): void {
```

Comment:
sorry for the nitpick, this threw me off when reviewing your PR :sweat:

Could you swap target and source?
I think it's standard that the target is first and then comes the source(/s)
like for Object.assign: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign